### PR TITLE
[500] Add application_open_from in the edit course form

### PR DESCRIPTION
**Pull request by stellenberger**
### Context

- [This trello ticket](https://trello.com/c/3QK8OQMI/500-allow-application-open-date-to-be-editable-in-the-support-console)

### Changes proposed in this pull request

- Allow support users to input a application form start date input in the SC
- Edit course details form to take dynamic date values to cut down code overhead

### Checklist

- [ ] Make sure all information from the Trello card is in here
- [ ] Attach to Trello card
- [ ] Rebased master
- [ ] Cleaned commit history
- [ ] Tested by running locally

### CODE REVIEW

**Comment by stellenberger** [comment]
[Allow application open date to be editable in the support console](https://trello.com/c/3QK8OQMI/500-allow-application-open-date-to-be-editable-in-the-support-console)

File: spec/forms/edit_course_form_spec.rb [hunk 1]
```diff
@@ -74,10 +75,11 @@ module Support
           subject.save
           subject.valid?
 
-          expect(subject.errors.messages.count).to eq(3)
+          expect(subject.errors.messages.count).to eq(4)
           expect(subject.errors.messages[:course_code]).to include("Course code cannot be blank")
           expect(subject.errors.messages[:name]).to include("Course title cannot be blank")
           expect(subject.errors.messages[:start_date]).to include("Start date cannot have blank values")
+          expect(subject.errors.messages[:applications_open_from]).to include("^Select when applications will open and enter the date if applicable")
```
Comment (defong) [inline spec/forms/edit_course_form_spec.rb:82]: Looks odd

**Review by defong — APPROVED** [review]
I think it good to go :muscle:

**Review by mbocc — APPROVED** [review]
All looks good! Error message also shows up when you put an incorrect applications open date.

Thank you!!

**Pull request merged**